File: oss/src/main/java/net/yan100/compose/oss/amazon/S3Args.java
package net.yan100.compose.oss.amazon;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class S3Args {
  @JsonProperty("Version")
  @SerializedName("Version")
  String version;

  @JsonProperty("Statement")
  @SerializedName("Statement")
  List<S3StatementArgs> statement = new ArrayList<>();
}

File: oss/src/main/kotlin/net/yan100/compose/oss/model/S3Args.kt
package net.yan100.compose.oss.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.gson.annotations.SerializedName

data class S3Args(
  @JsonProperty("Version")
  @SerializedName("Version")
  var version: String? = null,
  @JsonProperty("Statement")
  @SerializedName("Statement")
  var statement: MutableList<S3StatementArgs> = ArrayList(),
)
